src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferLengthCountIsEmptyOverAnyAnalyzer.cs

```diff
@@ -0,0 +1,142 @@
+        internal const string LengthText = nameof(Array.Length);
+        internal const string CountText = nameof(ICollection.Count);
+
+        internal const string IsEmptyId = "CA1860";
+        internal const string LengthId = "CA1861";
+        internal const string CountId = "CA1862";
+
+        private const string SourceParameterName = "source";
+
+        internal static readonly DiagnosticDescriptor IsEmptyDescriptor = DiagnosticDescriptorHelper.Create(
+            IsEmptyId,
+            CreateLocalizableResourceString(nameof(PreferIsEmptyOverAnyTitle)),
+            CreateLocalizableResourceString(nameof(PreferIsEmptyOverAnyMessage)),
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(PreferIsEmptyOverAnyDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false
+        );
+
+        internal static readonly DiagnosticDescriptor LengthDescriptor = DiagnosticDescriptorHelper.Create(
+            LengthId,
+            CreateLocalizableResourceString(nameof(PreferLengthOverAnyTitle)),
+            CreateLocalizableResourceString(nameof(PreferLengthOverAnyMessage)),
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(PreferLengthOverAnyDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false
+        );
+
+        internal static readonly DiagnosticDescriptor CountDescriptor = DiagnosticDescriptorHelper.Create(
+            CountId,
+            CreateLocalizableResourceString(nameof(PreferCountOverAnyTitle)),
+            CreateLocalizableResourceString(nameof(PreferCountOverAnyMessage)),
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(PreferCountOverAnyDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false
+        );
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterOperationAction(OnInvocationAnalysis, OperationKind.Invocation);
+        }
+
+        private static void OnInvocationAnalysis(OperationAnalysisContext context)
+        {
+            var invocation = (IInvocationOperation)context.Operation;
+            var originalMethod = invocation.TargetMethod.OriginalDefinition;
+            if (IsEligibleAnyMethod(originalMethod))
+            {
+                // The collection will be passed as the instance in VB and as an argument in C#. 
+                var firstArgument = invocation.Instance ?? invocation.Arguments[0].Value;
+                var type = (firstArgument as IConversionOperation)?.Operand.Type ?? firstArgument.Type;
```

Comment:
@mavasani Is this logic correct? The logic in `GetReceiverType` seems more expensive (involves semantic model calls).

for now, I think this should use `GetReceiverType`, and if this logic is correct and more efficient, then `GetReceiverType` should be updated as a follow up. What do you think?